mergin/client.py

```diff
@@ -134,6 +134,22 @@ def is_versioned_file(self, file):
         f_extension = os.path.splitext(file)[1]
         return f_extension in diff_extensions
 
+    def is_gpkg_open(self, path):
+        """
+        Check whether geopackage file is open (and wal file exists)
+
+        :param path: absolute path of file on disk
+        :type path: str
+        :returns: whether file is open
+        :rtype: bool
+        """
+        f_extension = os.path.splitext(path)[1]
+        if f_extension != '.gpkg':
+            return False
+        if os.path.exists(f'{path}-wal') and os.path.exists(f'{path}-shm'):
```

Comment:
Not sure if we should require also `-shm` file because it only provides shared memory access to multiple processes. Moreover the docs say that other means can be used to implement locking (instead of `-shm`). The uncommitted data we are interested in only live in the -wal file anyway...